diff --git a/libril/Android.mk b/libril/Android.mk
index ee0530f..d1e8ee3 100644
--- a/libril/Android.mk
+++ b/libril/Android.mk
@@ -1,5 +1,7 @@
 # Copyright 2006 The Android Open Source Project
 
+ifneq ($(BOARD_PROVIDES_LIBRIL),true)
+
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
@@ -48,3 +50,4 @@ LOCAL_SANITIZE := integer
 
 include $(BUILD_SHARED_LIBRARY)
 
+endif # BOARD_PROVIDES_LIBRIL
